### PR TITLE
Mh/createcluster

### PR DESCRIPTION
**Pull request by holmesm8**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This PR is for the curated cluster creation. Added new controller methods, a view page for the form, as well as verified that the new curated cluster name is displayed on the index page after creation.
## Related Tickets & Documents
N/A
## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
N/A
## Added tests?

- [x] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] Are there any post deployment tasks we need to perform?
No
## [optional] What gif best describes this PR or how it makes you feel?
Zapp Brannigan
alt_text

### CODE REVIEW

File: app/views/curated_clusters/index.html.erb [hunk 1]
```diff
@@ -1,5 +1,5 @@
-<%= @clusters %>
+<% @clusters.each do |cluster| %>
+  <%= cluster.name %>
+<% end %>
```
Comment (Yetidancer) [inline app/views/curated_clusters/index.html.erb:3]: good solution to our issue here!

File: app/views/curated_clusters/new.html.erb [hunk 2]
```diff
@@ -1 +1,10 @@
-POOP
```
Comment (Yetidancer) [inline app/views/curated_clusters/new.html.erb:1]: not sure why this needed to be removed

File: app/controllers/curated_clusters_controller.rb [hunk 3]
```diff
@@ -1,24 +1,27 @@
...
+  def create
+    @user = User.find_by(username: params["username"])
+    @user.curated_clusters.create(cluster_params)
+    redirect_to "/users/#{@user.username}/curated_clusters"
+  end
+
+  private
+
+  def cluster_params
+    params.permit(:name)
```
Comment (Yetidancer) [inline app/controllers/curated_clusters_controller.rb:25]: is this necessary

File: app/views/curated_clusters/index.html.erb (hunk 1 above)
Comment (ERaith) [reply app/views/curated_clusters/index.html.erb:3]: Looks good!

File: config/routes.rb [hunk 4]
```diff
@@ -412,10 +412,10 @@
   get "/:username/:slug" => "stories#show"
   get "/:username" => "stories#index"
 
-  #***
-  get '/users/:username/curated_clusters' => 'curated_clusters#index'
-  get '/users/:username/curated_clusters/:id' => 'curated_clusters#show'
-  post '/users/:username/curated_clusters/new' => 'curated_clusters#new'
+  get "/users/:username/curated_clusters" => "curated_clusters#index"
+  get "/users/:username/curated_clusters/new" => "curated_clusters#new"
+  get "/users/:username/curated_clusters/:id" => "curated_clusters#show"
```
Comment (ERaith) [inline config/routes.rb:417]: Nice catch on the wildcard here!

File: spec/features/clusters/new_spec.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+      user = create(:user)
+
+      visit "/users/#{user.username}/curated_clusters"
+      click_link "New Curated Cluster"
+      expect(page).to have_current_path("/users/#{user.username}/curated_clusters/new", ignore_query: true)
+
+      fill_in :name, with: "Matt's Cluster"
+      click_on "Create Curated Cluster"
+
+      expect(page).to have_current_path("/users/#{user.username}/curated_clusters")
```
Comment (Yetidancer) [inline spec/features/clusters/new_spec.rb:15]: why is "have_current_path" proper nomenclature? I haven't seen that before.

File: spec/features/clusters/new_spec.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
+require "rails_helper"
+
+RSpec.describe "cluster create", type: :feature do
+  describe "create page" do
+    it "curated cluster creation" do
+      user = create(:user)
+
+      visit "/users/#{user.username}/curated_clusters"
+      click_link "New Curated Cluster"
+      expect(page).to have_current_path("/users/#{user.username}/curated_clusters/new", ignore_query: true)
```
Comment (Yetidancer) [inline spec/features/clusters/new_spec.rb:10]: does the "ignore_query" mean ignore everything after a ?

File: spec/features/clusters/new_spec.rb (hunk 5 above)
Comment (holmesm8) [reply spec/features/clusters/new_spec.rb:15]: When attempting to commit, rubocop had some issues with the shoulda-matchers from before. The new syntax is "have_current_path" for verifying the page you should be on.

File: spec/features/clusters/new_spec.rb (hunk 6 above)
Comment (holmesm8) [reply spec/features/clusters/new_spec.rb:10]: only_path is deprecated and the new syntax is ignore_query

**Pull request merged**